reactor-core/src/main/java/reactor/core/publisher/FluxDematerialize.java

```diff
@@ -36,46 +32,32 @@
 
 	@Override
 	public void subscribe(CoreSubscriber<? super T> actual) {
-		source.subscribe(new DematerializeSubscriber<>(actual));
+		source.subscribe(new DematerializeSubscriber<>(actual, false));
 	}
 
-	static final class DematerializeSubscriber<T> extends AbstractQueue<T>
-			implements InnerOperator<Signal<T>, T>,
-			           BooleanSupplier {
+	static final class DematerializeSubscriber<T> implements InnerOperator<Signal<T>, T> {
 
 		final CoreSubscriber<? super T> actual;
+		final boolean isMono;
```

Comment:
Is it possible to change the name of this flag from "is of type" to "should be doing this"? I assume there isn't anything Mono specific here, just the behaviour flag.

Just thinking out loud of the future cases of having this behaviour for some another, non-Mono type

Reply:
this is quite `Mono` specific, but why not.